### ISSUE TITLE
Idle example hangs after running

### ISSUE THREAD

**Issue by anderspitman** [emersion/go-imap#455]
I'm not sure if this behavior is intentional. The Idle example hangs after it has received the first update. I believe the intended behavior is for it to exit. I was able to achieve that by changing this line:

https://github.com/emersion/go-imap/blob/16ffd077a2a8d66fca15b4b63b84a5504d2ccf4c/client/example_test.go#L295

to

```
updates := make(chan client.Update, 1)
```

Even if this is intentional, I have a question: I've been coding in Go for a couple years now, but I was surprised that an unserviced channel would result in the program blocking.

After this:

https://github.com/emersion/go-imap/blob/16ffd077a2a8d66fca15b4b63b84a5504d2ccf4c/client/example_test.go#L319

I would expect it to return and exit. Can someone explain this behavior or point me to a resource. I tried searching but don't know the right terms.

**Comment by vortura** [comment]
@anderspitman I ran into this problem too, so I thought I'd share my findings. The IMAP IDLE example assumes you already have a `*client.Client` to work with. For my test, I was connecting and logging in, then deferring the logout with `defer c.Logout()`.

When my client hung as you have observed, I obtained a stack dump with ctrl-\, and noticed two goroutines of interest. First, the deferred logout is waiting on a select here:
https://github.com/emersion/go-imap/blob/4bf026be1126813155f9f92ed8fe169e72d2cab5/client/client.go#L276

Second there's a goroutine trying to send to the updates channel here:
https://github.com/emersion/go-imap/blob/4bf026be1126813155f9f92ed8fe169e72d2cab5/client/client.go#L383

I believe what's happening is that as the function returns (this is all in `main()` for me), the deferred logout runs. This triggers an IMAP exchange, causing go-imap to try to send on the `c.Updates` channel. Because `c.Updates` is unbuffered in the example code and there's nothing to read on it, the call blocks. If you make the Updates channel buffered as you suggest, the code doesn't block and can proceed to exit.

Another option is to nil `c.Updates` before returning. If `Updates` is nil, the library doesn't send to it, so again the code can exit cleanly. e.g.

```
case err := <-done:
    if err != nil {
        log.Fatal(err)
    }
    log.Println("Not idling anymore")
    c.Updates = nil
    return
```

**Comment by emersion** [comment]
Closing because this is an issue about go-imap v1. I'm now focusing on go-imap v2.

Please re-open if you can reproduce with go-imap v2.
